Bottom: 5f2689fca0b3019d119d4c2a704cafec73cf205e
Top:    c3972894ac587422352563595bd7fbe982c7ae5e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 01:33:59 +0000

undo.c: Fix incorrect sizeof() when allocting memory for netlist undo

(Caught by coverity)

Coverity-cid: 241


---

diff --git a/src/undo.c b/src/undo.c
index 87e02aa..c3fc57e 100644
--- a/src/undo.c
+++ b/src/undo.c
@@ -1646,7 +1646,7 @@ AddNetlistLibToUndoList (LibraryTypePtr lib)
       undo->Data.NetlistChange.lib = lib;
 
       /* and what the old data is that we'll need to restore */
-      undo->Data.NetlistChange.old = (LibraryTypePtr)malloc (sizeof (LibraryTypePtr));
+      undo->Data.NetlistChange.old = (LibraryTypePtr)malloc (sizeof (LibraryType));
       old = undo->Data.NetlistChange.old;
       old->MenuN = lib->MenuN;
       old->MenuMax = lib->MenuMax;
